Loop/View Models/BolusEntryViewModel.swift

```diff
@@ -198,7 +201,26 @@ final class BolusEntryViewModel: ObservableObject {
                 presentAlert(.manualGlucoseEntryOutOfAcceptableRange)
                 return
             }
+        }
 
+        // Authenticate the bolus before saving anything
+        if enteredBolus.doubleValue(for: .internationalUnit()) > 0 {
+            let message = String(format: NSLocalizedString("Authenticate to Bolus %@ Units", comment: "The message displayed during a device authentication prompt for bolus specification"), enteredBolusAmountString)
+            authenticate(message) {
+                switch $0 {
+                case .success:
+                    self.continueSaving(onSuccess: completion)
+                case .failure:
+                    break
+                }
+            }
+        } else if let _ = potentialCarbEntry  { // Allow user to save carbs without bolusing
```

Comment:
[nit] Is this check necessary? How would we get here unless we had a carb entry with a zero bolus? I suppose it is safe to double check. Might want to use `else if potentialCarbEntry != nil {`.

Reply:
I didn't want to make assumptions about the fact we have a valid carb entry (we _should_, but the state of this view model is complicated enough that I figured a check wouldn't hurt). Checking if it's not nil is more straightforward, thanks!